sw/device/lib/dif/dif_aes.h

```diff
@@ -0,0 +1,589 @@
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occurred.
+   */
+  kDifAesReadOutputBadArg = kDifAesBadArg,
+  /**
+   * The AES unit has no valid output.
+   */
+  kDifAesReadOutputInvalid,
+} dif_aes_read_output_result_t;
+
+/**
+ * Reads AES Output Data.
+ *
+ * The peripheral must have finished previous encryption/decryption operation,
+ * and have valid data in the output registers (OUTPUT_VALID set), and will
+ * return `kDifAesReadOutputInvalid` if this condition is not met.
+ *
+ * @param aes AES state data.
+ * @param data AES Output Data.
+ * @return `dif_aes_read_output_result_t`.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aes_read_output_result_t dif_aes_read_output(const dif_aes_t *aes,
+                                                 dif_aes_data_t *data);
+
+/**
+ * AES Trigger flags.
+ */
+typedef enum dif_aes_trigger {
+  /**
+   * Trigger encrypt/decrypt.
+   */
+  kDifAesTriggerStart = 0,
+  /**
+   * Clear key, Initialisation Vector/Initial Counter Value and input data.
+   */
+  kDifAesTriggerKeyIvDataInClear,
+  /**
+   * Clear Output Data registers.
+   */
+  kDifAesTriggerDataOutClear,
+  /**
+   * Perform reseed of the internal state.
+   */
+  kDifAesTriggerPrngReseed,
+} dif_aes_trigger_t;
+
+/**
+ * Triggers one of `dif_aes_trigger_t` operations.
+ *
+ * All the triggers are applicable to both (automatic and manual) modes, with
+ * the exception of `kDifAesTriggerStart`, which is ignored in automatic mode.
+ *
+ * @param aes AES state data.
+ * @param trigger AES trigger.
+ * @return `dif_aes_result_t`.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aes_result_t dif_aes_trigger(const dif_aes_t *aes,
+                                 dif_aes_trigger_t trigger);
+
+/**
+ * AES Status flags.
+ */
+typedef enum dif_aes_status {
+  /**
+   * Device is idle.
+   */
+  kDifAesStatusIdle = 0,
+  /**
+   * Device has stalled (only relevant in automatic
+   * operation mode). Output data overwrite
+   * protection.
+   */
+  kDifAesStatusStall,
+  /**
+   * Device output is valid/ready. Denotes a
+   * successful encrypt or decrypt operation.
+   */
+  kDifAesStatusOutputValid,
+  /**
+   * Device Input Data registers can be written to
+   * (ready to accept new input data).
+   */
+  kDifAesStatusInputReady,
+  /**
+   * Fatal alert conditions include i) storage errors in the Control Register,
+   * and ii) if any internal FSM enters an invalid state.
+   */
+  kDifAesStatusAlertFatal,
+  /**
+   * Recoverable alert conditions include update errors in the Control Register.
+   */
+  kDifAesStatusAlertRecoverable,
+} dif_aes_status_t;
+
+/**
+ * Queries the AES status flags.
+ *
+ * @param aes AES state data.
+ * @param flag Status flag to query.
+ * @param set Flag state (set/unset).
+ * @return `dif_aes_result_t`.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_aes_result_t dif_aes_status_get(const dif_aes_t *aes, dif_aes_status_t flag,
```

Comment:
Hmmm, `get_status` for consistency with other DIFs?

Reply:
Done.